staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go

```diff
@@ -100,6 +103,12 @@ func newSimpleType(name string, exprType *exprpb.Type, zeroVal ref.Val) *DeclTyp
 	}
 }
 
```

Comment:
I don't think `NewTypeParam` is used anywhere, so I'll go ahead and remove it. `newSimpleType` is only used by `NullType` at the moment since I'm not sure what its serialized form is (after asking @jpbetz about it I very strongly suspect it's unused code; I've created #111328 to track removing it/other stuff from the copied CEL code we have).

I think setting `MinSerializedSize` everywhere is a good idea, yeah. I think the only place that it's not currently set in terms of used code is for maps/objects, so I'll go ahead and fix that.

Reply:
(resolving this since `NewTypeParam` is gone and `NullType` (and `UIntType`? I don't see any references to it giving the codebase a quick glance and grep) has a tracking issue, feel free to reopen)

Reply:
I still think we should make sure every DeclType has a non-zero MinSerializedSize. I'd be inclined to set NullType to size 4 and drop newSimpleType, but I'd also be ok with a MinSerializedSize of 1 (every json value requires at least one byte of serialized information to exist)

Something like this could resolve the remaining DeclType instances that don't have MinSerializedSize set:
```diff
diff --git a/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go b/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go
index b2b231f3ce5..ca74a50bfaf 100644
--- a/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go
+++ b/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/types.go
@@ -79,17 +79,13 @@ func NewObjectType(name string, fields map[string]*DeclField) *DeclType {
 	return t
 }
 
-func newSimpleType(name string, celType *cel.Type, zeroVal ref.Val) *DeclType {
-	return &DeclType{
-		name:         name,
-		celType:      celType,
-		defaultValue: zeroVal,
-	}
-}
-
 func newSimpleTypeWithMinSize(name string, celType *cel.Type, zeroVal ref.Val, minSize int64) *DeclType {
-	dt := newSimpleType(name, celType, zeroVal)
-	dt.MinSerializedSize = minSize
+	dt := &DeclType{
+		name:              name,
+		celType:           celType,
+		defaultValue:      zeroVal,
+		MinSerializedSize: minSize,
+	}
 	return dt
 }
 
@@ -558,7 +554,7 @@ var (
 	IntType = newSimpleTypeWithMinSize("int", cel.IntType, types.IntZero, minNumberSize)
 
 	// NullType is equivalent to the CEL 'null_type'.
-	NullType = newSimpleType("null_type", cel.NullType, types.NullValue)
+	NullType = newSimpleTypeWithMinSize("null_type", cel.NullType, types.NullValue, 4)
 
 	// StringType is equivalent to the CEL 'string' type which is expected to be a UTF-8 string.
 	// StringType values may either be string literals or expression strings.
diff --git a/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/value.go b/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/value.go
index 79817bead74..9a537aba97f 100644
--- a/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/value.go
+++ b/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/value.go
@@ -766,4 +766,4 @@ func celBool(pred bool) ref.Val {
 	return types.False
 }
 
-var unknownType = &DeclType{name: "unknown"}
+var unknownType = &DeclType{name: "unknown", MinSerializedSize: 1}
```

Reply:
Gotcha - I'll go ahead and apply those changes.